pkg/server/plugin/datastore/sql/sql_test.go

```diff
@@ -1456,14 +1456,51 @@ func (s *PluginSuite) TestListNodeSelectors() {
 
 		resp := s.listNodeSelectors(req)
 		s.Assert().Len(resp.Selectors, len(nonExpiredSelectorsMap))
-		for _, n := range resp.Selectors {
-			expectedSelectors, ok := nonExpiredSelectorsMap[n.SpiffeId]
+		for spiffeID, selectors := range resp.Selectors {
+			expectedSelectors, ok := nonExpiredSelectorsMap[spiffeID]
 			s.Assert().True(ok)
-			s.AssertProtoListEqual(expectedSelectors, n.Selectors)
+			s.AssertProtoListEqual(expectedSelectors, selectors)
```

Comment:
Could we do a similar check on line 1447 now that the return value is a map? The expected value `selectorMap` in the "list all" test function also organizes selectors by agent SPIFFE ID.

Reply:
Sure. Fixed all uses to go through a unified helper.